jax/interpreters/mlir.py

```diff
@@ -650,6 +660,7 @@ def lower_jaxpr_to_fun(
     jaxpr: the jaxpr to lower.
     effects: a sequence of `core.Effect`s corresponding to an ordering of tokens
       that will be created in or used by the lowered function.
+    create_tokens: if true, the MHLO will create tokens and ignore dummy input tokens.
```

Comment:
I wouldn't say "ignore", but rather something like "the computation will not expect input tokens nor emit output tokens for effects".

Implicitly, is there a TODO to remove this?

Reply:
I don't think so. Maybe this boolean should be renamed, but its purpose is to distinguish between the two possible lowerings of an effectful jaxpr:
1. It's a top-level jaxpr, meaning we need to create MHLO tokens and return the dummy tokens as output
2. It's a jaxpr in a higher-order-primitive, meaning we are directly passed MHLO tokens and just return them directly.

Reply:
Even for (1), isn't the intent that you pass runtime tokens eventually? (Absent some change to the runtime like another mechanism for ordering.)